runtime/service/src/main/java/org/apache/polaris/service/admin/PolarisPrincipalRolesServiceDefaultDelegator.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.polaris.service.admin;
+
+import jakarta.annotation.Priority;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.enterprise.inject.Alternative;
+import jakarta.enterprise.inject.Default;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.SecurityContext;
+import org.apache.polaris.core.admin.model.CreatePrincipalRoleRequest;
+import org.apache.polaris.core.admin.model.GrantCatalogRoleRequest;
+import org.apache.polaris.core.admin.model.UpdatePrincipalRoleRequest;
+import org.apache.polaris.core.context.RealmContext;
+import org.apache.polaris.service.admin.api.PolarisPrincipalRolesApiService;
+
+@RequestScoped
+@Default
+@EventsServiceDelegator
+@Alternative
+@Priority(1000) // Will allow downstream project-specific delegators to be added and used
+public class PolarisPrincipalRolesServiceDefaultDelegator
```

Comment:
nit: the naming convention is a bit odd, I'd expect the decorator to be named `<DelegateName>EventServiceDelegator` or something like that.

Reply:
So... what happens if we need delegators for something else? E.g. `PolarisDiagnostics` could be turned into another delegator, e.g. `<DelegateName>Diagnostics`.

The beauty of `@Decorator` is that you can in theory have as many decorators you need, and each time the decorated interface is injected, CDI would transparently create a chain of invocations that traverse all the decorators up to the "undecorated" bean. IOW if you inject `PolarisCatalogsApiService` somewhere, calling `PolarisCatalogsApiService.createCatalog()` would actually call `<DelegateName>Diagnostics.createCatalog()` then `<DelegateName>EventServiceDelegator.createCatalog()` then `PolarisServiceImpl.createCatalog()`.

Reply:
I understand what you are saying now - I think I was originally closer to Dmitri's idea earlier in this PR by attempting to use a delegator wrapper without the Decorator pattern. Let me put a general comment on the PR regarding this and change the naming convention to more accurately reflect the Decorator pattern.